### PR TITLE
Manage AWS SDK(s) version with Dependabot

### PR DESCRIPTION
**Pull request by zmoog**
## Proposed commit message

Set up Dependabot to manage the AWS SDK version.

With the current reactive and manual process, our dependencies are often outdated. To release a bugfix to a dependency, we need to wait for the following stack release instead of merging it shortly after it's available from AWS.

See https://github.com/elastic/beats/issues/39492 to learn more.

## Checklist

- [ ] ~~My code follows the style guidelines of this project~~
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [ ] ~~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Related issues

- Relates #39492

### CODE REVIEW

**Comment by zmoog** [comment]
The following dependencies are AWS-related, but I'm not sure they depend on the AWS SDK.

```
github.com/awslabs/kinesis-aggregation/go/v2
github.com/aws/smithy-go
github.com/awslabs/goformation/v4
github.com/aws/aws-lambda-go
```

**Comment by zmoog** [comment]
> [github.com/awslabs/kinesis-aggregation](https://github.com/awslabs/kinesis-aggregation/blob/master/go/v2/go.mod) depends on `github.com/aws/aws-sdk-go-v2/service/kinesis`

It seems Functionbeat only uses this package to [reference](https://github.com/elastic/beats/blob/2d687ac9d22854900d1ca76a4e22a2b1515a7886/x-pack/functionbeat/provider/aws/aws/transformer/transformer.go#L92-L98) the `types.Record` struct.

**Comment by kaiyan-sheng** [comment]
> It seems Functionbeat only uses this package to [reference](https://github.com/elastic/beats/blob/2d687ac9d22854900d1ca76a4e22a2b1515a7886/x-pack/functionbeat/provider/aws/aws/transformer/transformer.go#L92-L98) the types.Record struct.

Functionbeat is also using the deaggregator: https://github.com/elastic/beats/blob/main/x-pack/functionbeat/provider/aws/aws/transformer/transformer.go#L18

**Comment by zmoog** [comment]
> > It seems Functionbeat only uses this package to [reference](https://github.com/elastic/beats/blob/2d687ac9d22854900d1ca76a4e22a2b1515a7886/x-pack/functionbeat/provider/aws/aws/transformer/transformer.go#L92-L98) the types.Record struct.
> 
> Functionbeat is also using the deaggregator: https://github.com/elastic/beats/blob/main/x-pack/functionbeat/provider/aws/aws/transformer/transformer.go#L18

Oh, good catch!

We should probably initially focus on the AWS SDK only (`github.com/aws/aws-sdk-go-v2/*`), leaving other AWS-related packages for a later addition.

**Comment by zmoog** [comment]
We need to revise this PR after merging #40125

**Review by dliappis — APPROVED** [review]
LGTM (CI is green)

**Comment by zmoog** [comment]
I am waiting for https://github.com/elastic/beats/pull/40150 to merge before rebasing on it.

**Comment by zmoog** [comment]
I rebased on `main` to get the changes we made for Azure and GCP.

@rowlandgeoff, would you mind taking a final look at the AWS config?

**Comment by pazone** [comment]
Not sure if it's important, but we also use aws-lambda-go [here](https://github.com/elastic/beats/blob/5541971cc328e60ce9cc0b7408eb166cf0c58b0f/x-pack/functionbeat/provider/aws/aws/api_gateway_proxy.go#L12)

File: .github/dependabot.yml
Comment (zmoog) [reply .github/dependabot.yml:52]: Yep, it makes sense! I created separate PRs to facilitate this kind of conversation.

Do we want to group these libraries in the AWS SDK (also adding to `groups.aws-sdks.patterns`), or is it better to keep them separate (by only including them in the `allows` list)?

We grouped all the AWS SDK in a group with the `github.com/aws/aws-sdk-go-v2/*` pattern to avoid dependencies problems like https://github.com/elastic/beats/pull/39454 (a PR accidentally upgraded only the core library, breaking main).

**Comment by rowlandgeoff** [comment]
> I rebased on `main` to get the changes we made for Azure and GCP.
> 
> @rowlandgeoff, would you mind taking a final look at the AWS config?

Happy to help if you need to be unblocked, but will defer to engineers closer to the code. Based on the comments, looks like sign-off will wait for now.

**Comment by zmoog** [comment]
> Happy to help if you need to be unblocked, but will defer to engineers closer to the code. Based on the comments, looks like sign-off will wait for now.

Yeah, @rowlandgeoff sorry for the ping. The conversation got a new (and welcome) life right after I mentioned you. I'll request a new review when we complete the changes.

File: .github/dependabot.yml
Comment (zmoog) [reply .github/dependabot.yml:52]: I suggest adding github.com/aws/aws-lambda-go and github.com/aws/smithy-go as dependencies to check. If we are unsure now, we can still add them later to the group.

**Comment by zmoog** [comment]
@rowlandgeoff, I guess we're now ready for your review. We can make additional changes in future PRs as needed.

**Comment by rowlandgeoff** [comment]
LGTM, but I will defer to @dliappis since he's back from PTO

**Comment by zmoog** [comment]
Thanks! @dliappis, waiting for your feedback before merging the PR, then! 😇

**Comment by dliappis** [comment]
LGTM [still](https://github.com/elastic/beats/pull/39738#pullrequestreview-2165824005) from an eng prod PoV

**Pull request merged**
